### ISSUE TITLE
[a11y] Search engine logo in the search bar is missing accessible label

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#21207]
## Steps to reproduce
1. Ensure VoiceOver is running
2. Navigate to Fx New Tab page
3. Activate the search bar
4. Navigate to and focus with VoiceOver a search engine, i.e. `G` for Google, and observe the VO announcement

### Expected behavior
1. The name of the selected search engine is announced (the logo image has an accessible label), i.e. `Google`

### Actual behavior
1. The search engine's logo is exposed to VoiceOver as just `Image`.
2. Sometimes, VO tries to guess what is on the image, i.e. adding `Catio, G` after the step 1 announcement

### Device & build information
* Device: iPhone 15 Pro, iOS 17.5.1
* Build version: Fx Beta build 9000 (43786) (Nightly)
* First seen version: Fx release
### Notes
Attachments: 

![VoiceOver announcing G logo as Image](https://github.com/user-attachments/assets/603e39aa-2f5f-4733-b0f7-cdee6b491090)
![VoiceOver adding Catio G to the announcement after trying to analyze the image](https://github.com/user-attachments/assets/5033a89b-943d-44fd-8632-82ff16540fde)

**Comment by data-sync-user** [comment]
➤ Winnie Teichmann commented:

For QA: With the new toolbar being default now, this issue should be solved. Could you please verify?

**Comment by data-sync-user** [comment]
➤ Miruna Curtean commented:

Verified fixed with top and bottom toolbar with iPhone 11 (18.7) on v144.0(62506). 

!RecordIt-7E812C27-4E25-44B1-86E4-1756B79D8CD0.mp4|width=396,alt="RecordIt-7E812C27-4E25-44B1-86E4-1756B79D8CD0.mp4"!
